backend/elasticsearch/src/main/java/org/hibernate/search/backend/elasticsearch/search/projection/impl/ElasticsearchSearchProjection.java

```diff
@@ -6,10 +6,42 @@
  */
 package org.hibernate.search.backend.elasticsearch.search.projection.impl;
 
-import org.hibernate.search.backend.elasticsearch.search.extraction.impl.HitExtractor;
 import org.hibernate.search.engine.search.SearchProjection;
-import org.hibernate.search.engine.search.query.spi.ProjectionHitCollector;
+import org.hibernate.search.engine.search.query.spi.LoadingResult;
+import org.hibernate.search.engine.search.query.spi.ProjectionHitMapper;
 
-public interface ElasticsearchSearchProjection<T> extends SearchProjection<T>, HitExtractor<ProjectionHitCollector> {
+import com.google.gson.JsonObject;
 
+public interface ElasticsearchSearchProjection<T> extends SearchProjection<T> {
```

Comment:
Yup, I was hesitant because I thought it would be a mess but in the end, I think it's better than what we had.

Pushed an additional commit with it.